File: python_backend/pathsInfo/get_sub_path.py


def get_sub_path(path, i):
    """get up to and including the index i in path"""
    return path[:i + 1]

if __name__ == "__main__":
    path = get_sub_path(['#', 'WP1', 'WP2', 'BB1'], 1)
    print(path == ['#', 'WP1'])

File: app (1)/src/game_logic/pathsInfo/getSubPath.js
import {getStatuses} from "./getStatuses"


export function getSubPath(path, i) {
    /*get up to &&  including the index i in path**/
    return path.slice(0, i + 1)
}

// module.exports = getSubPath;

/** for node.js
if (require.main === module) {
    console.log(getSubPath(['#', 'WP1', 'WP2', 'BB1'], 1))
    console.log(getSubPath(['#', 'WP1', 'WP2', 'BB1'], 2))
    console.log(getSubPath(['#', 'WP1', 'WP2', 'BB1'], 3))

}
*/
